### DISCUSSION TITLE
Filtering notifications based on content

### DISCUSSION THREAD

**Discussion by erooke** [folke/noice.nvim#584] [Q&A] [answered]
Is it possible to filter notifications based on their contents? I am trying to stop the following error from displaying:

![image](https://github.com/folke/noice.nvim/assets/46689793/70c6f080-29f8-48cc-bba1-cc8b3c09015a)

The following filter does squash the notification but at the cost of every other notification
```lua
require("noice").setup({
	routes = {
		{
			filter = {
				event = "notify",
			},
			opts = { skip = true },
		},
	},
})

```
while the following filter fails to stop the notification
```lua
require("noice").setup({
	routes = {
		{
			filter = {
				event = "notify",
				find = "nil",
			},
			opts = { skip = true },
		},
	},
})
```
I have tried reading through the guide to messages and all the help documentation I could find but could not find how to filter this particular notification.

**Comment by Yolo390** [comment]
+1

**Comment by folke** [comment, marked as answer]
You can add `find = ...` to your filter

**Reply by Yolo390** [reply]
Thanks for your answer !

```lua
 │ │ routes = {
 │ │ │ -- Always route any messages with more than 15 lines to the split view.
 │ │ │ {
 │ │ │ │ view = "split",
 │ │ │ │ filter = { event = "notify", min_height = 15 },
 │ │ │ },
 │ │ │ -- WIP: try to disable Notify for Merginal plugin.
 │ │ │ {
 │ │ │ │ filter = {
 │ │ │ │ │ event = "notify",
 │ │ │ │ │ content = "Running command",
 │ │ │ │ },
 │ │ │ │ opts = {
 │ │ │ │ │ skip = true,
 │ │ │ │ },
 │ │ │ },
 │ │ },
```

It seems that `content` key not exist.

![Screenshot from 2023-10-04 13-28-50](https://github.com/folke/noice.nvim/assets/11678456/7eaed4d2-cc1e-4bf9-8afe-0669f53e242c)

I am trying to disable this notification from vim-merginal plugin.

![Screenshot from 2023-10-04 13-19-12](https://github.com/folke/noice.nvim/assets/11678456/d1499f51-b834-4203-ae7d-b83c4cbde9da)

**Reply by folke** [reply]
Sorry, should be `find`. It's all in the documentation https://github.com/folke/noice.nvim#-filters

**Reply by Yolo390** [reply]
Thanks, gonna try later.

Is there a better way to say, for this plugin (e.g. vim-merginal), do not display all notify messages ?

**Reply by folke** [reply]
nope, filters is the way to go. Unless you can of course just disable that notification in the plugin's options.

**Reply by Yolo390** [reply]
For this plugin, it seems not to have the possibility to disable notifications in options.
